### PR TITLE
Qemu

### PR DESCRIPTION
**Pull request by thebsdbox**
## Description

This adds the basic usage of Bare Metal with Qemu for the deployment hosts.

## Why is this needed

Adds more ways to use Tinkerbell !

Fixes: #

## How Has This Been Tested?

This has been used to implement new features in Tinkerbell !

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

File: docs/setup/using-qemu.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
+---
+title: Creating a virtual Tinkerbell environment with Qemu
+date: 2021-01-27
+---
+
+## pre-requisites
+
+This is a rough shopping list of skills/accounts that will be a benefit for this guide.
+
+- Equinix Metal portal account
```
Comment (gianarb) [inline docs/setup/using-qemu.md:10]: Is sit this needed? I presume everything that can run Qemu with a reasonable amount of memory and CPU is fine, I would like to change that in favor of something more generic

File: docs/setup/using-qemu.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+a70a54baff08        deploy_hegel_1         0.00%               2.984MiB / 7.79GiB   0.04%               0B / 0B             246kB / 0B          5
+33d953c9b057        deploy_db_1            13.66%              6.363MiB / 7.79GiB   0.08%               24.7MB / 22.2MB     1.5MB / 1.34GB      8
+2fa8a6064036        deploy_nginx_1         0.00%               2.867MiB / 7.79GiB   0.04%               341kB / 0B          0B / 0B             3
+```
+
+We can see that the components for the Tinkerbell stack are particularly light, with this in mind we can be very confident that we can have all of our userland components (tinkerbell/docker/bash etc..) within 1GB of ram and leave all remaining memory for the virtual machines. 
+
+That brings us onto the next part, which is how big should the virtual machines be?
+
+### In memory OS (OSIE)
```
Comment (gianarb) [inline docs/setup/using-qemu.md:34]: This should be: "Operating System installation environment" and I am not sure we need it, we can link to this guide https://docs.tinkerbell.org/services/osie/ and leave only what is required for this document

File: docs/setup/using-qemu.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+The **Action image** will be pulled locally from a repository and again written to disk, **however** the disk that these images are written to is a ram disk, so these images will again withhold available memory.
+
+Finally, these images when ran (**Action containers**) will have binaries in them that will require available memory in order to run.
+
+The majority of this memory usage from the as seen from above is for the in-memory filesystem in order to host the userland tools and the images listed in the workflow. From testing we've normally seen that **>2GB** is required, however if your workflow consists of large action images then this will need adjusting accordingly.
+
+With all this in consideration, we can use Equinix Metal machines T-Shirt sizes do determine the size of machine required. Given the minimal overhead for Tinkerbell and userland then a `t1.small.x86` (1CPU and 8GB or Ram), however if you're looking at deploying multiple machines with tinkerbell then ideally a machine with 32GB of ram will comfortably allow a comfortable amount of headroom.
+
+### Recomended Machine size or Equinix Metal instances and OS
```
Comment (gianarb) [inline docs/setup/using-qemu.md:53]: Move this to prerequisite and avoid specifying a concrete provider IMHO.

File: docs/setup/using-qemu.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+Connect to the VNC port with a client (the random port generated in this example is `6671`).. it will be exposed on the public address of our equinix metal host.
+
+Kill the VM:
+
+```
+shack vm stop --id f0cb3c -d
+```
+
+
+### Install sandbox dependencies
```
Comment (gianarb) [inline docs/setup/using-qemu.md:153]: I think this is almost the same guide I wrote here #66 I would like to merge a few of those commands there and link it to it from here (the fewest information we duplicate the easier documentation will be to maintain)

**Review by gianarb — COMMENTED** [review]
Great work!

**Review by gianarb — CHANGES_REQUESTED** [review]
This requires a bit of manipulation based on #66 and it has to be merged after that.

**Comment by thebsdbox** [comment]
Addressed your changes :-)  and fixed conflict

**Review by gianarb — APPROVED** [review]
Perfect!

**Review by DailyAlice — APPROVED** [review]

File: docs/setup/using-qemu.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+Check the inventory of your desired facility, but the recommended instances are below:
+
+- `c1.small.x86`
+- `c3.small.x86`
+- `x1.small.x86`
+
+For speed of deployment and modernity of the Operating System, either ubuntu 18.04 or ubuntu 20.04 are recommended.
+
+## Deploying Tinkerbell
```
[the remark this answers is not in the document]
Comment (thebsdbox) [reply docs/setup/using-qemu.md:67]: sure

File: docs/setup/using-qemu.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+```
+# Add Nginx address to Tinkerbell
+sudo ip addr add 192.168.1.2/24 dev plunder
+cd deploy
+source ../.env; docker-compose up -d
+```
+
+At this point we now have a server with available resource, we can create virtual machines and tinkerbell is listening on the correct internal network!
+
+## Create a workflow (debian example)
```
[the remark this answers is not in the document]
Comment (thebsdbox) [reply docs/setup/using-qemu.md]: Yeah, makes sense :)

**Pull request merged**